File: C/383_RansomNote.c
#include <stdbool.h>
bool canConstruct(char* ransomNote, char* magazine) {
    int alphabet[26] = {0};
    int i = 0;
    while(ransomNote[i] != '\0'){
        alphabet[ransomNote[i] - 'a']++;
        i++;
    }
    i = 0;
    while(magazine[i] != '\0'){
        alphabet[magazine[i] - 'a']--;
        i++;
    }
    for(int i = 0; i < 26; i++){
        if(alphabet[i] > 0){
            return false;
        }
    }
    return true;
}

File: C#/383_RansomNote.cs
public class Solution {
    public bool CanConstruct(string ransomNote, string magazine) {
        int[] alphabet = new int[26];
        int i;
        foreach(char c in ransomNote){
            i = magazine.IndexOf(c, alphabet[c - 'a']);
            if(i == -1){//character not in magazine
                return false;
            }
            alphabet[c-'a'] = i+1;
        } 
        return true;
    }
}